Makefile

```diff
@@ -127,69 +100,187 @@ config: ## Set up default config
 
-fmt: headers ## Format the entire code base(s)
-	@./hack/code-format
+.PHONY: clean-gen
+clean-gens: ## Clean gen directories
+	@rm -rf aurae-proto/src/gen/*
+	@rm -rf auraescript/gen/*
+
+#------------------------------------------------------------------------------#
+
+# Protobuf Commands
 
 .PHONY: proto
-proto: proto-lint ## Generate code from protobuf schemas
+proto: proto-lint ## Lint and Generate code from protobuf schemas
 	@buf --version >/dev/null 2>&1 || (echo "Warning: buf is not installed! Please install the 'buf' command line tool: https://docs.buf.build/installation"; exit 1)
 	buf generate -v api
 
-.PHONY: cri
-cri: ## Copy the CRI interface from upstream
-	curl https://raw.githubusercontent.com/kubernetes/cri-api/$(cri_version)/pkg/apis/runtime/v1/api.proto -o api/kubernetes/cri/v1/$(cri_version).proto
-
 .PHONY: proto-lint
 proto-lint: ## Lint protobuf schemas
 	buf lint api
 
-.PHONY: clean
-clean: clean-certs
-	@rm -rf aurae-proto/src/gen/*
-	@cargo clean
+.PHONY: proto-vendor-cri
+proto-vendor-cri: ## Copy the CRI interface from upstream
+	curl https://raw.githubusercontent.com/kubernetes/cri-api/$(cri_version)/pkg/apis/runtime/v1/api.proto -o api/kubernetes/cri/v1/$(cri_version).proto
 
-headers: headers-write ## Fix headers. Run this if you want to clobber things.
+#------------------------------------------------------------------------------#
 
-headers-check: ## Only check for problematic files.
-	./hack/headers-check
+# Auraed Commands
 
-headers-write: ## Fix any problematic files blindly.
-	./hack/headers-write
+.PHONY: auraed
+auraed: proto auraed-lint auraed-debug ## Lint and install auraed (for use during development)
 
-.PHONY: spawn
-spawn: ## Spawn the current auraed binary and start it in a container
-	./hack/spawn
+.PHONY: auraed-lint
+auraed-lint:
+	@$(cargo) clippy --target $(uname_m)-unknown-linux-musl -p auraed --all-features -- -D clippy::all -D warnings
 
-.PHONY: busybox
-busybox: ## Creat a "busybox" OCI bundle in target
-	./hack/oci-busybox
+.PHONY: auraed-test
+auraed-test:
+	@$(cargo) test --target $(uname_m)-unknown-linux-musl -p auraed
 
-.PHONY: alpine
-alpine: ## Creat an "alpine" OCI bundle in target
-	./hack/oci-alpine
+.PHONY: auraed-test-all
+auraed-test-all:
+	@sudo -E $(cargo) test --target $(uname_m)-unknown-linux-musl -p auraed -- --include-ignored
 
-.PHONY: start
-start:
-	sudo -E $(HOME)/.cargo/bin/auraed
+.PHONY: auraed-debug
+auraed-debug:
+	@$(cargo) install --target $(uname_m)-unknown-linux-musl --path ./auraed --debug --force
 
-.PHONY: help
-help:  ## Show help messages for make targets
-	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(firstword $(MAKEFILE_LIST)) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[32m%-30s\033[0m %s\n", $$1, $$2}'
+.PHONY: auraed-release
+auraed-release:
+	@$(cargo) install --target $(uname_m)-unknown-linux-musl --path ./auraed --force
+
+#------------------------------------------------------------------------------#
+
+# AuraeScript Commands
```

Comment:
maybe bring back the auraescript/Makefile and then:

```
TARGETS := all clean lint test test-all debug release
SUBDIRS := $(wildcard */.)

$(TARGETS): $(SUBDIRS)
$(SUBDIRS):
        $(MAKE) -C $@ $(MAKECMDGOALS)

.PHONY: $(TARGETS) $(SUBDIRS)
```

this would allow "make test" to iterate through all the subdirectories calling "make test" in each.

Reply:
I'm not sure I 100% understand this, so I'll ask that we put it off for another PR if we want it.

Reply:
sorry, i got a bit carried away.  that's fine, i can do this later.  it's an optimization to make this all a bit cleaner.